src/code/z_actor.c

```diff
@@ -3397,6 +3397,10 @@ Actor* func_80033684(PlayState* play, Actor* explosiveActor) {
  * This is done by moving it to the corresponding category list and setting its category variable accordingly.
  */
 void Actor_ChangeCategory(PlayState* play, ActorContext* actorCtx, Actor* actor, u8 actorCategory) {
+    //! @bug If Actor_ChangeCategory is called during an actor update function, the loop in
+    //! Actor_UpdateAll will continue from the next actor in the new category, rather than the next
+    //! actor in the old category. This might cause some actors in the old category to be skipped
+    //! over and not updated, or some actors in the new category to be updated twice.
```

Comment:
Just some slight rewording suggestions. 
The first sentence I think just helps the reader understand what is the cause of the problem. And I think its more encompassing to say "an actor that is currently updating" rather than "the update function", since it can be from any function called by the actors update function
```
Calling this function immediately moves an actor from one category list to the other. This can cause a problem if called from an actor that is currently updating. 
The loop in `Actor_UpdateAll` will continue from the next actor in the new category, rather than the next
actor in the old category. Depending on the category the actor who called this function was moved to, this can either cause some actors in the old category to skip updating on that frame, or some actors in the new category to be updated twice.
```

Reply:
Remove "either", it can (and typically would) be both

Reply:
This can cause whole categories to be skipped as well right? might be worth rephrasing that whole part in general

Reply:
it can't skip categories because the outer loop should iterate over every category; Actor_ChangeCategory just breaks the link for the category currently being processed.

Reply:
If I remember correctly the different relevant bits of code, an actor A moving from category B to category C means that

- any actor after A in B will not be updated (the recursive iterating on actors inside categories is from head to tail)
- and actors in C will be updated instead (actors added to categories are added to the head of the linked list so that would be all previous C actors). C actors will also be updated during the normal iteration for C (either before or after B)

If B is before C, A gets updated twice (first as part of B with the category switch, then as part of C once the iteration for C comes), otherwise A is only updated as part of B

-----

Personally I don't think this bug is major enough to warrant long explanations, just a short description would do and one can figure out the details.

Reply:
also, in theory it's possible to execute actors more than twice per cycle. For example, let's say you have Actor A in category 1 moving to category 3, actor B in category 2 also moving into category 3, and Actor C in category 3

* Actor A will be updated twice
* Actor B will be updated once
* Actor C will be updated three times.

Reply:
Okay, I added the first sentence that Fig suggested, and made the wording about the effects a bit more precise.

(I disagree about needing to expand to "an actor that is currently updating", typically I count anything that happens during a function called from `f` as also happening during `f`, such as when asking "how long does this function take to run")

Reply:
dunno, I think its better to describe the process of updating rather than referring to a specific function that happens to be the entry point for carrying it out

Reply:
meh I would understand them both the same so I think it's better to be more concise

Reply:
The way I phrased it is not much longer and it leaves less room for confusion, it just seems better all around.. If someone else wants to weigh in real quick to tie break this, would be appreciated

Reply:
Alright, changed to "actor update"